### DISCUSSION TITLE
How to get local ip address？

### DISCUSSION THREAD

**Discussion by ningan123** [cue-lang/cue#1799] [Q&A]
There is my test.cue
```
# test.cue

ip: ????
```

When I run
`cue export test.cue`

I want to get a local ip address 
`ip: 192.168.11.11`

Could anyone have some good ideas?

**Comment by slewiskelly** [comment]
This would either have to be injected, retrieved/filled using the scripting layer or Go API. CUE's configuration layer is hermetic and this is the only way to inject values from the environment.

> I want to get a local ip address

I don't know if this can be solved in a generic way, a system may have many interfaces each with an IP address, which one is chosen, and how to support various platforms?

**Comment by mvdan** [comment]
We can add built-in support for more system variables alongside current ones like `os` and `cwd`, such as https://github.com/cue-lang/cue/issues/1692. I believe we're OK with adding more as long as they are reasonably clear and commonly used.

I'm not sure that "local IP" meets that criteria, like @slewiskelly says, but you can always inject external variables yourself, such as:

```
$ cat foo.cue 
ip: string @tag(ip)
$ cue export foo.cue 
ip: incomplete value string:
    ./foo.cue:1:5
$ cue export -t ip=$(hostname --ip-address) foo.cue 
{
    "ip": "127.0.1.1"
}
```

Also, I'm moving this to the Q&A section of the discussions tab, as it's a question more than a feature request :)
